components/GraphDiagram/graph_util.ts

```diff
@@ -142,9 +142,10 @@ export async function getGraphForQuery(
         const downstreamModule = await getModule(getModuleKey(name, version));
 
         const moduleInfo = await _visit(downstreamModule, level + 1);
+        if (!moduleInfo) return;
 
-        moduleInfo?.upstream.add({ module, type });
-        info?.downstream.add({ module: downstreamModule, type });
+        moduleInfo.upstream.add({ module, type });
+        info.downstream.add({ module: moduleInfo.module, type });
```

Comment:
The main change here is `module: moduleInfo.module` so that if `moduleInfo` was [already cached](https://github.com/npmgraph/npmgraph/blob/ceb40b728a064654842d43ec92641c9dec24a8bf/components/GraphDiagram/graph_util.ts#L118-L123), we reuse its module instance. Otherwise there is multiple instance of `Module` in the graph, which makes the `seen: Set<Module>` cache below not working correctly.

Reply:
This was a workaround for a bug that was allowing a module to have more than one entry in the cache.  I've just added the fix for that (`newCachedModule()`), and reverted this.